### PR TITLE
ci: add linting, code coverage and other useful checks

### PR DESCRIPTION
**Pull request by shibumi**
I don't know if this is useful, but I invested a few minutes and build a proper release pipeline + added tests with golangci.
If some of these steps are already covered by our jenkins pipeline I can remove them.

### CODE REVIEW

**Comment by shibumi** [comment]
converting this to a WIP/draft... I want to make a few adjustments:

1. provide documentation for the pipeline
2. edit Makefile, so it is consistent with the pipeline. @georgettica  suggested that we could use goreleaser for building local binaries, too.

**Comment by georgettica** [comment]
Odd, I can't add emojis to these comments.
But I agree and think this is a good way to go

File: .goreleaser.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+project_name: cadctl
+builds:
+  - ldflags:
+      - "-s -w"
+      - "-extldflags=-zrelro"
```
Comment (georgettica) [inline .goreleaser.yaml:5]: you had comments before on zrelro being a requirement for stuff, can you add it again?

Comment (shibumi) [reply .goreleaser.yaml:5]: Oh yeah.. I don't know where they went

Comment (shibumi) [reply .goreleaser.yaml]: done

**Comment by shibumi** [comment]
@georgettica  what do we do with this PR here? Now, that we know that the GitHub pipeline won't work GoReleaser will not work as well (50% of the goreleaser config file only works in combination with github actions).

I would suggest we remove the goreleaser/github workflow part and just add the codecov hack files we need for code coverage.

I will try to have a look on prow (openshift/release). Would be cool if we could at least add `golangci-lint` as Linter for our Go code.

**Comment by georgettica** [comment]
Sadly I agree.

I think we can also ask if we can add this GHA as it hardens our binaries, but we will see about that later on

**Comment by shibumi** [comment]
@georgettica  well, the binary hardening can be added in the makefile as well. These are just go flags :)

**Comment by georgettica** [comment]
Then this can be done aswell fairly easy

**Comment by shibumi** [comment]
@georgettica Done. I have incorporated the changes from GoReleaser into the Makefile. Will add make targets for golangci-lint gosec etc next.

**Comment by shibumi** [comment]
@georgettica this adds make targets for golangci-lint + gosec and removes the goreleaser files again.

Do we need a separate target for installing the lint/test dependencies? (running go install {gosec|golangci-lint})?

**Comment by georgettica** [comment]
I think gosec was not officially noted as something we need (have a discussion on this still)

but I think golangci-lint is a good addition

**Review by georgettica — CHANGES_REQUESTED** [review]
One we merge this we can retest the release code

**Comment by georgettica** [comment]
@shibumi could you make all of these commands pass?
I want to merge this so the CI will verify our stuff :)

**Comment by georgettica** [comment]
it seems go test also needs the vendor thingy XP

**Comment by shibumi** [comment]
/retest


Triggering a retest because the job is pending for over 20min now..

**Pull request merged**